test/test_charm.py

```diff
@@ -468,6 +471,46 @@ def _get_action_test_meta(cls):
   additionalProperties: false
 ''')
 
+    def test_meta_from_charm_root(self):
+        with tempfile.TemporaryDirectory() as d:
+            td = pathlib.Path(d)
+            (td / 'metadata.yaml').write_text(
+                yaml.safe_dump(
+                    {"name": "bob",
+                        "requires": {
+                            "foo":
+                                {"interface": "bar"}
+                        }}))
+
+            meta = ops.CharmMeta.from_charm_root(td)
+            assert meta.name == "bob"
+            assert meta.requires['foo'].interface_name == "bar"
```

Comment:
We're keen on generally moving to pytest, largely to get assertions in this style, but I think while we're still in unittest, we should probably stick with the unittest assertion methods for the most part.

```suggestion
            self.assertEqual(meta.name, "bob")
            self.assertEqual(meta.requires['foo'].interface_name, "bar")
```